### ISSUE TITLE
[Bug] Visual Map display Error

### ISSUE THREAD

**Issue by 2796673723** [apache/echarts#18386]
### Version

5.4.0

### Steps to Reproduce

I want to set different colors inside and outside of a range by using `visualMap.` Examples options are as follows:
```javascript
{
  show: false,
  pieces: [
    {
      gte: 0,
      lte: limit,
      color: '#4A90E2',
    },
  ],
  outOfRange: { color: '#FF0000' },
}
```

### Current Behavior

The line at the bottom appears in two colors.
![image](https://user-images.githubusercontent.com/43328314/225247123-1a9ceeb8-f3e8-4143-b9c7-f03ba5a2c0d5.png)

### Expected Behavior

As per the settings I want the bottom line to only show blue.

### Environment

```markdown
- OS:Windows10
- Browser:Chrome 110.0.5481.180
- Framework:Vue@2.7
```

**Comment by helgasoft** [comment]
yes, a bug on pieces border values, [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AMYCGYVA5sAE7m0A0J6AE2aMiAbVoBZGL1i0AKgFcqM2gHUqAlXIAWClQDEOEFQGVmphdFoBdEgF8-6cjnxFUpMpRqyAbowA2SrT2jrB4VEZUhLCi_MQegsJiAIwArAAMPLDJAEyZ2XlZAMz5pVnp1qEeFNR0_hDQylWk9Y0mFP7eSLAA7hACYNpEabB2cWPolSQ--AoBEowgbnF42sA9RABmAeHNIBBU9FFicejuCeisLET5APS3EGCw2ox4ZNpUYYwAtp8cUQp_E8eh84ApwrIrsELuggcNCqdSPRgP5OHRWP8qFZYPcAEQAYgALFgAJzpACiOVxzXQE1IUw8wAUYAA8psAEqMaCsLqwZGojh0fH6fTpMXpWijewoOwAbiAA)
![image](https://user-images.githubusercontent.com/13038071/225523575-9c505e0b-1314-4187-8a43-db36e151f87a.png)

**Comment by symlons** [comment]
It seems like that VisualMap isn't accounting for the LineWidth.

**Comment by Steve245270533** [comment]
This problem is the same as what I encountered. Have you resolved it in the end?

**Comment by day-xue** [comment]
为什么还没解决啊

**Comment by Ovilia** [comment]
@Steve245270533 @day-xue This is an open-source project and if you want your issues to be fixed faster, the best way is to make a pull request rather than waiting for others to help.

**Comment by akwangl** [comment]
他为什么不设计成指定的数组下标来设置颜色
